megatron/data/t5_dataset.py

```diff
@@ -195,7 +234,8 @@ def pad_and_convert_to_numpy(tokens, masked_positions,
     # Decoder-side padding mask.
     num_tokens_dec = len(t5_decoder_in)
     padding_length_dec = max_seq_length_dec - num_tokens_dec
-    assert padding_length_dec >= 0
+    if padding_length_dec < 0:
+        raise DecoderLengthExceededError()
```

Comment:
`LengthExceededError` and `DecoderLengthExceededError` can possibly happen quite often I guess? Why not truncate the sequences to the max length instead?

Reply:
The `LengthExceededErrors` do happen often if the model's max sequence length is chosen too low. This is the same behavior as in the `T5Dataset`; it just does not happen often here because the amount of spans (and – correspondingly – inserted `extra_id` tokens) is much lower. For the `UL2Dataset`, I had [previously added code that calculated a truncation of the sequence based on the maximum number of possible maskings](https://github.com/NVIDIA/Megatron-LM/pull/268/files/324d70d2518fe183c27e36ed550a95955eaa077c#diff-982756bf3108b36f634314abf1c582d2f44e34b84450edab4b93c9f82579b3d6R162-R167), but decided that raising an error is probably the better, user-facing way of handling this problem.